### PR TITLE
Admin endpoints

### PR DESCRIPTION
**Pull request by MartinezNicanor**

### CODE REVIEW

File: src/routes/admin.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+router.post('/login', (req: Request, res : Response) => postLoginAdmin(req, res));
+
+// import endpoints that require a token
+router.use(authenticateToken)
+
+router.get('/profile/:id', (req: Request, res : Response) => getAdminProfile(req, res));
+
+router.get('/juniorView', (req: Request, res : Response) => getJuniorView(req, res));
+
+router.get('/juniorView', (req: Request, res : Response) => getMediorView(req, res));
```
Comment (ZsomborHajzer) [inline src/routes/admin.ts]: Change to /mediorView

File: src/routes/admin.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// import endpoints that require no token
+router.post('/login', (req: Request, res : Response) => postLoginAdmin(req, res));
+
+// import endpoints that require a token
+router.use(authenticateToken)
+
+router.get('/profile/:id', (req: Request, res : Response) => getAdminProfile(req, res));
+
+router.get('/juniorView', (req: Request, res : Response) => getJuniorView(req, res));
```
Comment (ZsomborHajzer) [inline src/routes/admin.ts]: add middleware routes to juniorView, mediorView and seniorView like : 

router.get('/juniorView', allowJuniorMediorSenior,(req: Request, res : Response) => getJuniorView(req, res));

File: src/controller/admin.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        responder(res, 200, 'data', view)
+        return;
+    } catch(err) {
+        responder(res, 500, 'error', 'Internal Server Error');
+        return;
+    }
+}
+
+export const getMediorView = async (req:Request & { user?: User }, res:Response) => {
+    if (req.user?.user_type !== 'Medior') {
```
Comment (ZsomborHajzer) [inline src/controller/admin.ts]: if you apply middlewares on the routs directly (explained in comments for routes/admin file) this if statement  can be deleted

File: src/controller/admin.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+        responder(res, 200, 'data', view)
+        return;
+    } catch(err) {
+        responder(res, 500, 'error', 'Internal Server Error');
+        return;
+    }
+}
+
+export const getSeniorView = async (req:Request & { user?: User }, res:Response) => {
+    if (req.user?.user_type !== 'Senior') {
```
Comment (ZsomborHajzer) [inline src/controller/admin.ts]: same here if statement can be deleted if middlewares are applied

File: src/controller/admin.ts [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        responder(res, 200, 'profile', profile);
+        return;
+    } catch(err) {
+        responder(res, 500, 'error', 'Internal Server Error');
+        return;
+    }
+};
+
+export const getJuniorView = async (req:Request & { user?: User }, res:Response) => {
+    if (req.user?.user_type !== 'Junior') {
```
Comment (ZsomborHajzer) [inline src/controller/admin.ts]: if you apply middlewares on the routs directly (explained in comments for routes/admin file) this if statement can be deleted

**Review by ZsomborHajzer — CHANGES_REQUESTED** [review]
these changes i commented on are necessary

File: src/controller/admin.ts (hunk 3 above)
Comment (ZsomborHajzer) [reply src/controller/admin.ts]: the comment explaining  this is the one above

**Review by JoeyKraemer — APPROVED** [review]

**Review by JoeyKraemer — DISMISSED** [review]

**Comment by JoeyKraemer** [comment]
LGTM!

**Review by ZsomborHajzer — APPROVED** [review]

**Pull request merged**